Source/CLI/Global.cpp

```diff
@@ -0,0 +1,285 @@
+        OutputOptions["threads"] = argv[i];
+        return 0;
+    }
+
+    return Error_NotTested(argv[i]);
+}
+
+//---------------------------------------------------------------------------
+int global::ManageCommandLine(const char* argv[], int argc)
+{
+    if (argc < 2)
+        return Usage(argv[0]);
+
+    AttachementMaxSize = (size_t)-1;
+    DisplayCommand = false;
+
+    for (int i = 0; i < argc; i++)
+    {
+        if ((argv[i][0] == '.' && argv[i][1] == '\0')
+         || (argv[i][0] == '.' && (argv[i][1] == '/' || argv[i][1] == '\\') && argv[i][2] == '\0'))
+        {
+            //translate to "../xxx" in order to get the top level directory name
+            char buff[FILENAME_MAX];
+            getcwd(buff, FILENAME_MAX);
+            string Arg = buff;
+            size_t Path_Pos = Arg.find_last_of("/\\");
+            Arg = ".." + Arg.substr(Path_Pos);
+            Inputs.push_back(Arg);
+        }
+        else if ((strcmp(argv[i], "--attachment-max-size") == 0 || strcmp(argv[i], "-s") == 0) && i + 1 < argc)
+            AttachementMaxSize = atoi(argv[++i]);
+        else if ((strcmp(argv[i], "--bin-name") == 0 || strcmp(argv[i], "-b") == 0) && i + 1 < argc)
+        {
+            int Value = SetBinName(argv[++i]);
+            if (Value)
+                return Value;
+        }
+        else if ((strcmp(argv[i], "--display-command") == 0 || strcmp(argv[i], "-d") == 0))
+        {
+            int Value = SetDisplayCommand();
+            if (Value)
+                return Value;
+        }
+        else if (strcmp(argv[i], "--help") == 0 || strcmp(argv[i], "-h") == 0)
+            return Help(argv[0]);
+        else if (strcmp(argv[i], "--version") == 0)
+            return Version();
+        else if ((strcmp(argv[i], "--output-file-name") == 0 || strcmp(argv[i], "-o") == 0) && i + 1 < argc)
+        {
+            int Value = SetOutputFileName(argv[++i]);
+            if (Value)
+                return Value;
+        }
+        else if ((strcmp(argv[i], "--rawcooked-file-name") == 0 || strcmp(argv[i], "-r") == 0) && i + 1 < argc)
+            rawcooked_reversibility_data_FileName = argv[++i];
+        else if (argv[i][0] == '-' && argv[i][1] != '\0')
+        {
+            int Value = SetOption(argv, i, argc);
+            if (Value)
+                return Value;
+        }
+        else
+            Inputs.push_back(argv[i]);
+    }
+
+    return 0;
+}
+
+//---------------------------------------------------------------------------
+void global::SetDefaults()
+{
+    // Container format
+    if (OutputOptions.find("f") == OutputOptions.end())
+        OutputOptions["f"] = "matroska"; // Container format is Matroska
+
+    // Video format
+    if (OutputOptions.find("c:v") == OutputOptions.end())
+        OutputOptions["c:v"] = "ffv1"; // Video format is FFV1
+
+    // Audio format
+    if (OutputOptions.find("c:a") == OutputOptions.end())
+        OutputOptions["c:a"] = "copy"; // Audio is copied

```

Comment:
There is no change here compared to previous code.
FLAC decoder is not yet integrated to RAWcooked revert code, so it will be changed to FLAC (with option for PCM) when I integrate it.